ledger_secure_sdk_sys/build.rs

```diff
@@ -173,42 +342,37 @@ impl SDKBuilder {
             ),
         };
         self.device = device;
+        // export TARGET into env for 'infos.rs'
+        println!("cargo:rustc-env=TARGET={}", self.device.to_string());
         println!("cargo:warning=Device is {:?}", self.device);
     }
 
-    /// Manually retrieve API_LEVEL from an SDK in the case of
-    /// path given through the LEDGER_SDK_PATH env variable
-    fn retrieve_api_level(bolos_sdk: &Path) -> Result<u32, SDKBuildError> {
-        let makefile_defines = File::open(bolos_sdk.join("Makefile.defines"))
-            .expect("Could not find Makefile.defines");
-        for line in BufReader::new(makefile_defines).lines().flatten() {
-            if line.contains("API_LEVEL") {
-                return line.split(":=").collect::<Vec<&str>>()[1]
-                    .trim()
-                    .parse()
-                    .map_err(|_| SDKBuildError::InvalidAPILevel);
-            }
-        }
-        Err(SDKBuildError::CouldNotGetAPILevel)
-    }
-
     pub fn bolos_sdk(&mut self) -> Result<(), SDKBuildError> {
         println!("cargo:rerun-if-env-changed=LEDGER_SDK_PATH");
-        let (bolos_sdk, api_level) = match env::var("LEDGER_SDK_PATH") {
-            Err(_) => clone_sdk(&self.device),
-            Ok(path) => {
-                let sdkpath = Path::new(&path).to_path_buf();
-                let apilevel = SDKBuilder::retrieve_api_level(&sdkpath)?;
-                (sdkpath, apilevel)
-            }
+        let sdk_info = match env::var("LEDGER_SDK_PATH") {
+            Err(_) => clone_sdk(&self.device)?,
+            Ok(path) => retrieve_sdk_info(&self.device, path)?,
         };
-        self.bolos_sdk = bolos_sdk;
-        self.api_level = api_level;
+        self.bolos_sdk = sdk_info.bolos_sdk;
+        self.api_level = sdk_info.api_level;
 
-        // export API_LEVEL into env for 'infos.rs'
+        // Export SDK infos into env for 'infos.rs'
         println!("cargo:rustc-env=API_LEVEL={}", self.api_level);
+        println!(
+            "cargo:rustc-env=API_LEVEL_STR={}",
+            format!("{}", self.api_level)
+        );
         println!("cargo:warning=API_LEVEL is {}", self.api_level);
-
+        println!("cargo:rustc-env=TARGET_ID={}", sdk_info.target_id);
+        println!("cargo:warning=TARGET_ID is {}", sdk_info.target_id);
+        println!("cargo:rustc-env=TARGET_NAME={}", sdk_info.target_name);
+        println!("cargo:warning=TARGET_NAME is {}", sdk_info.target_name);
+        println!("cargo:rustc-env=C_SDK_NAME={}", sdk_info.c_sdk_name);
+        println!("cargo:warning=C_SDK_NAME is {}", sdk_info.c_sdk_name);
```

Comment:
In the deployer script, we use only the sdk_name field, where we put `"Rust SDK + " "ledger-secure-sdk"`. It ends in the yml artifact and in the tag pushed on the repo.
I think there is no great reason to have a different behavior, though I'm not sure which version we should choose.